### ISSUE TITLE
Local builds, prerequisites and 

### ISSUE THREAD

**Issue by radu-matei** [metaparticle-io/package#50]
At the moment, all builder implementation require Docker client locally.

Has it been discussed the possibility of having something similar to Draft that builds and pushes your image without you having anything locally?

Thanks!

**Comment by brendandburns** [comment]
Yes, I would love to see support for this!

The 'builder' interface is factored out in each runtime to enable this kind of extensibility.

PRs welcome :)

**Comment by radu-matei** [comment]
Ok, so in this case:

- there will be a pod in the Kubernetes cluster that will create and push the image
- based on that image, the local `mp-compiler` will push the deployment on the cluster.

Is this brief overview correct? Do you have anything else in mind?
(Maybe the `mp-compiler` also as a service in the cluster?)

Now, do we also want to enable support for custom a Dockerfile? (Simply as an optional param as path to a local Dockerfile). The way I see from the runtimes, the Dockerfile is created on the fly, and the user doesn't have the ability to have any input anything.

Thanks!

**Comment by brendandburns** [comment]
Yeah, sounds about right. mp-compiler is actually built as a RESTful server too, so it can definitely run in a cluster.

--brendan

________________________________________

**Comment by stealthybox** [comment]
Related #49
